app/utils/metadata.py

```diff
@@ -569,506 +569,309 @@ def __update_from_settings(self) -> None:
-                                    self.internal_local_metadata[uuid],
-                                    load_these_after_by_ver["li"],
-                                    "loadTheseAfter",
-                                    "loadTheseBefore",
-                                    self.internal_local_metadata,
-                                    self.packageid_to_uuids,
-                                )
-                            else:
-                                logger.warning(
-                                    f"About.xml syntax error. Unable to read <loadbeforebyversion> tag from XML for version [{version}]: {self.internal_local_metadata[uuid]['metadata_file_path']}"
-                                )
-                                logger.debug(load_these_after_by_ver)
-                        except Exception as e:
-                            mod_metadata_path = self.internal_local_metadata[uuid].get(
-                                "metadata_file_path"
+                        if isinstance(load_before_by_ver, dict):
+                            load_before_by_ver_list = (
+                                load_before_by_ver.get("li")
+                                if load_before_by_ver
+                                else None
+                            )
+                            process_load_order(
+                                uuid,
+                                load_before_by_ver_list,
+                                "loadTheseBefore",
+                                "loadTheseAfter",
                             )
+                        else:
                             logger.warning(
-                                f"Error processing <loadbeforebyversion> tag for {version} from XML: {mod_metadata_path}"
+                                f"Invalid format for 'loadafterbyversion' data for mod {uuid}: {load_before_by_ver}"
                             )
-                            logger.debug(e)
 
-        logger.info("Finished adding dependencies through About.xml information")
+            # Process loadbefore
+            load_before = mod_metadata.get("loadbefore", {})
+            load_before_list = load_before.get("li") if load_before else None
+            process_load_order(
+                uuid, load_before_list, "loadTheseAfter", "loadTheseBefore"
+            )
+
+            # Process forceloadbefore
+            force_load_before = mod_metadata.get("forceloadbefore", {})
+            force_load_before_list = (
+                force_load_before.get("li") if force_load_before else None
+            )
+            process_load_order(
+                uuid, force_load_before_list, "loadTheseAfter", "loadTheseBefore"
+            )
+
+            # Process loadbeforebyversion
+            load_before_by_version = mod_metadata.get("loadbeforebyversion", {})
+            if load_before_by_version and isinstance(load_before_by_version, dict):
```

Comment:
Just use static type checking if the param is supposed to only be one type

Reply:
Fixed